mrp_production_force_bom/models/mrp_production.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    def _check_unexpected_products(
+        self, move_raw_product_ids, bom_product_ids, error_messages
+    ):
+        unexpected_products = move_raw_product_ids - bom_product_ids
+        if unexpected_products:
+            error_messages.append(
+                _("'%s': Some components are not part of the BOM.") % self.name
+            )
+        return error_messages
+
+    def _check_missing_products(
+        self, move_raw_product_ids, bom_product_ids, error_messages
+    ):
+        missing_products = bom_product_ids - move_raw_product_ids
+        if missing_products:
+            error_messages.append(
+                _(
+                    "'%s': Some components from the BOM are missing"
+                    " in the raw materials."
+                )
+                % self.name
+            )
+        return error_messages
+
+    @api.model
+    def _force_bom(self):
+        param = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("mrp_production_force_bom.force_bom")
+        )
+        if param == "1":
+            return True
+        return False
+
+    @api.model
+    def _skip_bom_check(self):
+        """
+        Extend this method as necessary to control whether or not to check the components
+        against BoM.
+        """
```

Comment:
`@api.model` should not be added here.
```suggestion
    def _skip_bom_check(self):
        """Extend this method as necessary to control whether or not to check the
        components against BoM.
        """
```

Reply:
Yes, you right. I confused.